### PR TITLE
Convert more loops to comprehensions in metric.py

### PR DESCRIPTION
**Pull request by eric-wieser**
I'd argue that by reducing the number of local variables, this makes things more readable

### CODE REVIEW

**Comment by utensil** [comment]
> **new alerts:**
> 
> * 1 for Unused local variable

image

**Comment by eric-wieser** [comment]
Hopefully the flake8 fix in #279 will mean that type of mistake is caught immediately.

File: galgebra/metric.py [hunk 1]
```diff
@@ -690,37 +691,38 @@ def __init__(
...
+                        ]
+                        for coord in self.coords
+                    ])
+                    self.g_inv = Matrix([
+                        [
+                            Function('g__{}__{}'.format(coord, coord2))(*self.coords)
+                            for coord2 in self.coords
+                        ]
+                        for coord in self.coords
+                    ])
```
Comment (eric-wieser) [inline galgebra/metric.py:725]: These lines aren't covered

Comment (eric-wieser) [reply galgebra/metric.py:725]: `g = Ga('e*1|2', g='g', coords=sympy.symbols('x y'))` exercises it and doesn't crash, although it seems `g.g_inv` gets overwritten anyway.

**Pull request merged**